core/state/statedb.go

```diff
@@ -832,8 +832,8 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 }
 
 // IntermediateRoot computes the current root hash of the state trie.
-// It is called in between transactions to get the root hash that
-// goes into transaction receipts.
+// It is called in between blocks to get the root hash that
+// goes into block header.
```

Comment:
The comment here was correct. The method is used before the Byzantium fork blocks to return the hash after every tx.

Reply:
I think it was correct before, but is wrong now.
Due to it's meaningful for old blocks, so I would not insist on it.
Thanks for your reply!